fix: load sequences in unit page

ihor-romaniuk opened the pull request:
## Description
Handle loading errors when opening the course unit page via direct link as an unauthorized user.

## Backport:
- teak branch https://github.com/openedx/frontend-app-authoring/pull/1876

## Screenshots
Case 1: Open course unit page
![image](https://github.com/user-attachments/assets/758f8810-fadc-46ce-9959-97ecfb97cc40)

Case 2: Open Split Test xblock content
<img width="777" alt="image" src="https://github.com/user-attachments/assets/03b0dbe3-b2b6-4522-a6fe-ada8bdc5c91d" />
```
Unexpected Application Error!
Cannot read properties of undefined (reading 'upstreamLink')
TypeError: Cannot read properties of undefined (reading 'upstreamLink')
    at T2 (https://app.pr-1788-ba9a65.sandboxes.opencraft.hosting/authoring/app.a1ce6d6e6800b33ff46c.js:1:5941058)
    at pa (https://app.pr-1788-ba9a65.sandboxes.opencraft.hosting/authoring/457.db97dc1cceb0d252ed30.js:2:1924360)
    at ks (https://app.pr-1788-ba9a65.sandboxes.opencraft.hosting/authoring/457.db97dc1cceb0d252ed30.js:2:1938398)
    at wu (https://app.pr-1788-ba9a65.sandboxes.opencraft.hosting/authoring/457.db97dc1cceb0d252ed30.js:2:1984168)
    at gl (https://app.pr-1788-ba9a65.sandboxes.opencraft.hosting/authoring/457.db97dc1cceb0d252ed30.js:2:1972553)
    at hl (https://app.pr-1788-ba9a65.sandboxes.opencraft.hosting/authoring/457.db97dc1cceb0d252ed30.js:2:1972481)
    at pl (https://app.pr-1788-ba9a65.sandboxes.opencraft.hosting/authoring/457.db97dc1cceb0d252ed30.js:2:1972344)
    at nl (https://app.pr-1788-ba9a65.sandboxes.opencraft.hosting/authoring/457.db97dc1cceb0d252ed30.js:2:1969179)
    at tl (https://app.pr-1788-ba9a65.sandboxes.opencraft.hosting/authoring/457.db97dc1cceb0d252ed30.js:2:1967735)
    at _ (https://app.pr-1788-ba9a65.sandboxes.opencraft.hosting/authoring/457.db97dc1cceb0d252ed30.js:2:1005861)
```

## Testing instructions
- Open the course unit page directly while not signed in.
- Sign in to the system.
- Case 1
-- Verify that the course unit page loads correctly after authentication.
- Case 2
-- Add Split Test xBlock
-- Open Split Test xBlock content page
-- Verify that the Split Test xBlock content page loads correctly.

## Other information
**Settings**
```
EDX_PLATFORM_REPOSITORY: https://github.com/openedx/edx-platform.git
EDX_PLATFORM_VERSION: master

TUTOR_GROVE_WAFFLE_FLAGS:
  - name: contentstore.new_studio_mfe.use_new_unit_page
    everyone: true

TUTOR_GROVE_NEW_MFES:
  authoring:
    port: 18000
    repository: https://github.com/raccoongang/frontend-app-course-authoring.git
    version: romaniuk/fix/load-sequences

TUTOR_GROVE_MFE_LMS_COMMON_SETTINGS: |
  MFE_CONFIG["ENABLE_UNIT_PAGE"] = True
  MFE_CONFIG["ENABLE_NEW_EDITOR_PAGES"] = True
```

openedx-webhooks commented:
Thanks for the pull request, @ihor-romaniuk!


This repository is currently maintained by `@openedx/2u-tnl`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>

<summary>:radio_button: Update the status of your PR</summary>

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.
</details>
<!-- comment:end_of_wip -->


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IHRydWV9 -->

openedx-webhooks commented:
Thanks for the pull request, @ihor-romaniuk!


This repository is currently maintained by `@openedx/2u-tnl`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>

<summary>:radio_button: Update the status of your PR</summary>

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.
</details>
<!-- comment:end_of_wip -->


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IHRydWV9 -->

bradenmacdonald commented:
> Open the course unit page directly while not signed in.

I think you are missing the root cause here.

When I open a unit page directly and then log in, without accessing other Authoring MFE pages first, I see this in the network logs:
* http://studio.local.openedx.io:8001/api/contentstore/v1/container_handler/block-v1:A+B+C+type@vertical+block@… ✅
* http://studio.local.openedx.io:8001/xblock/container/block-v1:A+B+C+type@vertical+block@… ❌ 302 REDIRECT to /login

The second request fails because the `studio_session_id` cookie is not set, and [the `/xblock/container/:id` endpoint](https://github.com/openedx/edx-platform/blob/1d95a56dd8bd87bd6de56f947be9a40a7a1c7aaa/cms/djangoapps/contentstore/views/block.py#L374-L377) is not a "proper" REST API endpoint - it doesn't use DRF and doesn't accept auth headers, only cookies.

So really, I think we need to fix this properly - either:
1) Remove all calls to the old `/xblock/container/` endpoint and use `/api/contentstore/v1/container_handler/` everywhere instead - this will also speed up loading because we don't call two similar APIs but just one. Or:
2) Fix the old `/xblock/container/` endpoint so that it accepts header authentication. But I'd strongly prefer option 1 because the old `/xblock/container/` endpoint is also very confusing - if you open it in your browser without sending the right JSON headers, it shows a 404 error even though it's a valid URL.

It seems like there's also duplicate requests being made to each of these endpoints, so in the long term we really should be using React Query to load the data since it will de-duplicate the requests better.

bradenmacdonald commented on src/course-unit/course-sequence/hooks.js:
```diff
@@ -12,15 +12,15 @@ export function useSequenceNavigationMetadata(courseId, currentSequenceId, curre
   const sequenceIds = useSelector(getSequenceIds);
   const sequenceIndex = sequenceIds.indexOf(currentSequenceId);
-  const unitIndex = sequence.unitIds.indexOf(currentUnitId);
+  const unitIndex = sequence.unitIds?.indexOf(currentUnitId);
```
Adding `?` means that `unitIndex` can be `undefined`, and in that case we shouldn't be adding to it below on line 21. If you convert this to TypeScript it would show this as an error:

![Screenshot showing error](https://github.com/user-attachments/assets/d43b5b39-822c-4a84-89fe-fab68d0410aa)

bradenmacdonald commented:
@navinkarkera @rpenido @pomegranited FYI on this PR/error.

arbrandes commented:
What @bradenmacdonald said:

> we're just going to get more errors like this until we fix the root cause

What I suggest is for a new issue to be created with the relevant info, and for us to merge this one as a workaround for Teak in the meantime.

ihor-romaniuk commented:
@arbrandes @bradenmacdonald Thanks for your feedback.

I have replaced the old `/xblock/container/ `endpoint with the new `/api/contentstore/v1/container_handler/`.
It helps us to remove duplicate requests and avoid errors which I fixed initially.

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-1867-ba9a65.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-1867-ba9a65.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9961895332-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9961895332-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9961895332-tutor-requirements.txt)

arbrandes commented:
@ihor-romaniuk, I still see the error (or _an_ error) when I go to the unit page directly while logged out, then immediately log in:

![image](https://github.com/user-attachments/assets/322ee180-1197-4a71-8258-6ca5c871358c)

This is in this PR's sandbox, btw.

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9977713690.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9977713690-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9977713690-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9977713690-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9979307674.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9979307674-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9979307674-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9979307674-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9982262997.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9982262997-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9982262997-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9982262997-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9982576998.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9982576998-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9982576998-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9982576998-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-1867-ba9a65.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-1867-ba9a65.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9982262968-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9982262968-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9982262968-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-1867-ba9a65.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-1867-ba9a65.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9984485356-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9984485356-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9984485356-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-1867-ba9a65.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-1867-ba9a65.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9985201728-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9985201728-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9985201728-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-1867-ba9a65.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-1867-ba9a65.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9987001957-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9987001957-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9987001957-tutor-requirements.txt)

ihor-romaniuk commented:
@arbrandes I’ve finally resolved the issue with the instance.

The problem was caused by deploying from the wrong repository. We were deploying from the RG space frontend-app-course-authoring, but no changes were visible because the MFE ID was set to course-authoring instead of authoring.

I added a custom configuration to the sandbox settings, and everything is now working as expected:
```
TUTOR_GROVE_NEW_MFES:
  authoring:
    port: 18000
    repository: https://github.com/raccoongang/frontend-app-course-authoring.git
    version: romaniuk/fix/load-sequences
```

Please take a look when you get a chance.

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-1867-ba9a65.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-1867-ba9a65.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9989063078-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9989063078-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-9989063078-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-10017406091.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-10017406091-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-10017406091-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-10017406091-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-1867-ba9a65.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-1867-ba9a65.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-10017406074-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-10017406074-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-10017406074-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-1867-ba9a65.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-1867-ba9a65.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-10057035328-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-10057035328-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-10057035328-tutor-requirements.txt)

pomegranited commented:
Hi @ihor-romaniuk , could you add a bit of test coverage to fix the code coverage?

bradenmacdonald commented:
@arbrandes Would you be able to take another look at this soon? I think it's a pretty high priority bugfix. Or do you need someone else to take on the review?

And yes, @ihor-romaniuk , we'll need the codecov check passing to be able to merge this.

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-1867-ba9a65.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-1867-ba9a65.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-10170520900-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-10170520900-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-1867-ba9a65-51564656-10170520900-tutor-requirements.txt)

ihor-romaniuk commented:
@pomegranited @bradenmacdonald @arbrandes Test coverage was updated and it's ready for review.

bradenmacdonald reviewed (approved):
Nice! It's looking good and it fixes the bugs that I have seen. Since it's such a big refactor I'm not sure if we should backport it to Teak though. Perhaps just the 

![Screenshot 2025-05-28 at 12 01 09 PM](https://github.com/user-attachments/assets/b07e4877-7ecb-43bc-a0e0-1e85a258ba24)

part, or some smaller bit like that is what we backport?

Edit: that's done - https://github.com/openedx/frontend-app-authoring/pull/2042

bradenmacdonald commented on src/course-unit/data/api.js:
```diff
@@ -55,7 +40,10 @@ export async function getCourseSectionVerticalData(unitId) {
+  courseSectionVerticalData.xblockInfo.readOnly = isUnitReadOnly(courseSectionVerticalData.xblockInfo);
+
+  return courseSectionVerticalData;
 }
```
@ihor-romaniuk What do you think about renaming this before we merge?

bradenmacdonald replied:
Nice, thanks!

bradenmacdonald commented:
@arbrandes would you be able to take a quick look here too before we merge, or no?
